pkg/aws/profiles.go

```diff
@@ -0,0 +1,103 @@
+package aws
+
+import (
+	"github.com/go-ini/ini"
+	"github.com/mitchellh/go-homedir"
+	"path/filepath"
+)
+
+// MapProfile gets an AWS profile from the user's credentials file and maps it,
+// if found, to the given interface
+func (a *Aws) MapProfile(name string, profile interface{}) error {
```

Comment:
should profile be the Profile struct?  not a generic interface?  Seems like since that is what we are wanting here we should enforce it otherwise we will just write whatever interface into the creds file.

Reply:
So, I had done this originally however I wanted the ability to pass additional unknown fields as part of the profile. In this case the Vault lease ID. Since I didnt want Vault inherently part of the aws pkg I went with a generic interface. Might be a better way, not sure...

Reply:
yeah that gets tough with a struct, I would say for now we should enforce what is written for now.